### DISCUSSION TITLE
Using brew edit then brew update seems to mess with git status

### DISCUSSION THREAD

**Discussion by rdp** [Homebrew/discussions#359] [Everyday usage]
Hello.

I'm attempting to modify a formula to prepare "a pull request as you usually do"
https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md "To contribute a fix to the foo formula" section.

However,  it seems that git gets in a funky state, which makes this tricky, ex:

```
$  cd $(brew --repository homebrew/core) 
$ pwd
 /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
# all is well...
$ brew edit freerdp
# make a change...
$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   Formula/freerdp.rb
# all is normal git-land
# now wait awhile for a change to happen to some other formula.  Or if you don't want to wait for it, simulate it: $ git stash && git reset --hard HEAD~~ && git stash pop
$ brew update
Updated 1 tap (homebrew/core).
No changes to formulae.
$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        deleted:    .github/ISSUE_TEMPLATE.md
        deleted:    .github/ISSUE_TEMPLATE/bug.md
        deleted:    .github/ISSUE_TEMPLATE/config.yml
        ....
# it shows everything as deleted.  Hard to view git status/diffs now if I happen to run update after an edit...brew update still functions OK and pulls in latest, but always shows as deleted...
```

I'm not sure what is going on...Is this worth reporting?  It makes doing PR's "as I normally do" confusing at times...Thanks.

**Comment by rdp** [comment]
Also this does the same thing, making it pretty hard to avoid...

```
$ brew edit freerdp
# edit it
$ brew install --build-from-source freerdp # a part of the instructions for doing a PR
==> Auto-updated Homebrew!
Updated 3 taps (homebrew/cask-versions, homebrew/core and homebrew/cask).
```

Also for followers this has the following behavior:
```
$ git diff 
# empty output, even though the formula has been edited...
```

**Reply by rdp** [reply]
Though appears you can instruct brew not to update, for followers: https://github.com/Homebrew/brew/issues/1670

**Comment by Rylan12** [comment]
Hmm, this is odd. Not sure exactly what could be causing this. I can't reproduce locally (followed your instructions step-by-step. Can you post the output of `brew config` and `brew doctor`? Maybe there's something there that can give a clue.

Also, is `brew update` actually deleting all of those files, or does `git` just think that they're deleted?

One thing you can try is to run `brew update-reset`. This will reset all of the tap git repositories. **This is destructive, though, so make sure you have no uncommitted _or_ committed changes in _any_ taps or in the homebrew repository before doing this.**
